src/hevm/src/EVM.hs

```diff
@@ -834,36 +833,36 @@ exec1 = do
                              (initialContract (EVM.RuntimeCode mempty))
                              . storage . at x . non 0)
 
-                    let cost =
-                          if (current == new) then g_sload
-                          else if (current == original) && (original == 0) then g_sset
-                          else if (current == original) then g_sreset
-                          else g_sload
+                    let cost
+                          | (current == new) = g_sload
+                          | (current == original) && (original == 0) = g_sset
+                          | (current == original) = g_sreset
+                          | otherwise = g_sload
```

Comment:
```suggestion
                    let cost | (current == new) = g_sload
                                 | (current == original) && (original == 0) = g_sset
                                 | (current == original) = g_sreset
                                 | otherwise = g_sload
```